tests/components/lifx/test_sensor.py

```diff
@@ -0,0 +1,196 @@
+        data={CONF_HOST: IP_ADDRESS},
+        unique_id=MAC_ADDRESS,
+    )
+    config_entry.add_to_hass(hass)
+    bulb = _mocked_hev_bulb()
+    with _patch_discovery(device=bulb), _patch_config_flow_try_connect(
+        device=bulb
+    ), _patch_device(device=bulb):
+        await async_setup_component(hass, lifx.DOMAIN, {lifx.DOMAIN: {}})
+        await hass.async_block_till_done()
+
+    expected = {
+        "sensor.my_bulb_clean_cycle_duration": ("7200", TIME_MINUTES),
+        "sensor.my_bulb_clean_cycle_remaining": ("0", TIME_MINUTES),
+        "sensor.my_bulb_clean_cycle_last_power": ("False", None),
+        "sensor.my_bulb_clean_cycle_last_result": ("unknown", None),
+    }
+
+    entity_id = "light.my_bulb"
+    state = hass.states.get(entity_id)
+    assert state.state == "off"
+
+    for sensor_entity_id, value in expected.items():
+        state = hass.states.get(sensor_entity_id)
+        assert state.state == value[0]
+        assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == value[1]
+
+    updated = {
+        "sensor.my_bulb_clean_cycle_duration": "7200",
+        "sensor.my_bulb_clean_cycle_remaining": "300",
+        "sensor.my_bulb_clean_cycle_last_power": "True",
+    }
+
+    for sensor_entity_id, value in updated.items():
+        await async_update_entity(hass, sensor_entity_id)
+        await hass.async_block_till_done()
+
+        state = hass.states.get(sensor_entity_id)
+        assert state.state == value
+
+    entity_registry = er.async_get(hass)
+    disabled_entity = entity_registry.async_update_entity(
+        "sensor.my_bulb_clean_cycle_last_power",
+        disabled_by=er.RegistryEntryDisabler.INTEGRATION,
+    )
+    assert disabled_entity.disabled
+
+
+async def test_failing_sensors(hass: HomeAssistant) -> None:
+    """Test the RSSI sensor on a LIFX bulb."""
+    config_entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={CONF_HOST: IP_ADDRESS},
+        unique_id=MAC_ADDRESS,
+    )
+    config_entry.add_to_hass(hass)
+    bulb = _mocked_hev_bulb()
+    with _patch_discovery(device=bulb), _patch_config_flow_try_connect(
+        device=bulb
+    ), _patch_device(device=bulb):
+        await async_setup_component(hass, lifx.DOMAIN, {lifx.DOMAIN: {}})
+        await hass.async_block_till_done()
+
+    entity_registry = er.async_get(hass)
+    entities = [
+        entity
+        for entity in async_entries_for_config_entry(
+            entity_registry, config_entry.entry_id
+        )
+        if entity.domain == Platform.SENSOR
+    ]
+
+    for entity in entities:
+        entity_registry.async_update_entity(entity.entity_id, disabled_by=None)
+
+    with _patch_discovery(device=bulb), _patch_config_flow_try_connect(
+        device=bulb
+    ), _patch_device(device=bulb):
+        await hass.config_entries.async_reload(config_entry.entry_id)
+        await hass.async_block_till_done()
+
+    lifx_coordination: LIFXCoordination = hass.data[DOMAIN][config_entry.entry_id]
```

Comment:
We don't want to access `hass.data` in tests since thats considered  integration details.

patch the library instead and call async_fire_time_changed

https://developers.home-assistant.io/docs/development_testing?_highlight=writing&_highlight=tests#writing-tests-for-integrations

Reply:
I'm not sure I'm following. Can you point me towards an existing example of this, perhaps?